### ISSUE TITLE
[check package] Object can't have the "nodeType" property

### ISSUE THREAD

**Issue by jagi** [meteor/meteor#7354]
Because of this line https://github.com/meteor/meteor/blob/devel/packages/check/isPlainObject.js#L36 it is not possible to check existence of the `nodeType` property. It treats it as non plain object. The following code will not work.

``` js
const pattern = Match.ObjectIncluding({ nodeType: Match.Any });
check({ nodeType: 'asd' }, pattern);
// Throws "Match error: Expected plain object"
```

**Comment by hwillson** [comment]
The current `isPlainObject` function in the `check` package was extracted from jquery 1.11.2, to remove the `check` packages dependency on `jquery`. Modern versions of jquery's [`isPlainObject`](https://github.com/jquery/jquery/blob/3e3b09d68d317a3fe567e377e9032319ac5e0829/src/core.js#L236-L255) function do not have the `nodeType` issue mentioned in this issue thread ([example](https://jsfiddle.net/xkq7gnsv/)). We can likely just drop a copy of jquery's updated `isPlainObject` function into the `check` package (after massaging it a bit and adding some additional tests), and be good to go on this one.

**Comment by abernix** [comment]
`check@1.2.5` was just published with the fix submitted in #8351.  Thanks for reporting this originally, @jagi, thanks to @hwillson for the analysis and thanks @nlhuykhang for the fix!  Great cooperation!
